game/lib/login/login_page.dart

```diff
@@ -0,0 +1,123 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:flutter_signin_button/flutter_signin_button.dart';
+import 'package:google_fonts/google_fonts.dart';
+import 'package:game/api/game_api.dart';
+import 'package:game/home_page/home_page_widget.dart';
+
+class LoginWidget extends StatefulWidget {
+  final FlutterSecureStorage storage;
+  final String API_URL;
+  LoginWidget({Key? key, required this.storage, required this.API_URL})
+      : super(key: key);
+
+  @override
+  _LoginWidgetState createState() => _LoginWidgetState();
+}
+
+class _LoginWidgetState extends State<LoginWidget> {
+  final scaffoldKey = GlobalKey<ScaffoldState>();
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance?.addPostFrameCallback((_) {
+      //to auto log in a user if already logged in.
+      checkLoggedIn();
+    });
+  }
+
+  Future<void> checkLoggedIn() async {
+    if (await ApiClient(widget.storage, widget.API_URL).tryRelog()) {
+      _toHomePage(context);
+    }
+  }
+
+  void _toHomePage(context) {
+    Navigator.pop(context);
+    Navigator.push(
+        context, MaterialPageRoute(builder: (context) => HomePageWidget()));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        key: scaffoldKey,
+        backgroundColor: Colors.black,
+        body: Center(
+            child: Column(
+          mainAxisAlignment: MainAxisAlignment.spaceAround,
+          children: [
+            Column(
+              children: [
+                Container(
+                    child: Image.asset('assets/images/logo_hires.png'),
+                    width: 200,
+                    height: 200),
+                Text("CornellGO!",
+                    style: GoogleFonts.lato(
+                        textStyle: TextStyle(
+                            color: Colors.white,
+                            fontWeight: FontWeight.bold,
+                            fontSize: 35))),
+                Padding(
+                  padding: const EdgeInsets.only(top: 8.0),
+                  child: Text("INTERACTIVE SCAVENGER HUNT",
+                      style: GoogleFonts.lato(
+                          textStyle:
+                              TextStyle(color: Colors.white, fontSize: 14))),
+                )
+              ],
+            ),
+            Column(
+              children: [
+                Padding(
+                  padding: const EdgeInsets.only(bottom: 30),
+                  child: SignInButton(
+                    Buttons.Google,
+                    onPressed: () async {
+                      final client = ApiClient(widget.storage, widget.API_URL);
```

Comment:
You should not be creating a new ApiClient, but consuming the one at the top of the hierarchy. Use the consumer widget for that.